langstream-agents/langstream-agent-s3/src/main/java/ai/langstream/agents/s3/S3Processor.java

```diff
@@ -0,0 +1,194 @@
+        String region = configuration.getOrDefault("region", "").toString();
+        String objectName = configuration.getOrDefault("objectName", "").toString();
+
+        // Object name is a mustache template because it is passed in the record
+        objectTemplate = Mustache.compiler().compile(objectName);
+
+        log.info(
+                "Connecting to S3 Bucket at {} in region {} with user {}",
+                endpoint,
+                region,
+                username);
+
+        MinioClient.Builder builder =
+                MinioClient.builder().endpoint(endpoint).credentials(username, password);
+        if (!region.isBlank()) {
+            builder.region(region);
+        }
+        minioClient = builder.build();
+    }
+
+    @Override
+    public void process(List<Record> records, RecordSink recordSink) {
+        for (Record record : records) {
+            processRecord(record, recordSink);
+        }
+    }
+
+    @Override
+    public ComponentType componentType() {
+        return ComponentType.PROCESSOR;
+    }
+
+    private void processRecord(Record record, RecordSink recordSink) {
+        log.debug("Processing record {}", record.toString());
+        MutableRecord context = MutableRecord.recordToMutableRecord(record, true);
+        final JsonRecord jsonRecord = context.toJsonRecord();
+
+        String fileName = objectTemplate.execute(jsonRecord);
+
+        log.info("Processing file {}", fileName);
+
+        // Retrieve headers from the original record
+        Collection<Header> originalHeaders = record.headers();
+
+        log.debug("Original headers: {}", originalHeaders);
+
+        try {
+            GetObjectArgs getObjectArgs =
+                    GetObjectArgs.builder().bucket(bucketName).object(fileName).build();
+
+            GetObjectResponse getObjectResponse = minioClient.getObject(getObjectArgs);
+
+            // Read the file content from the response
+            byte[] fileContent = getObjectResponse.readAllBytes();
+            log.debug("File content: {}", new String(fileContent));
+
+            // Create a new record with the file content
+            Record processedRecord = new S3SourceRecord(fileContent, fileName, originalHeaders);
+            log.debug("Processed record key: {}", processedRecord.key());
+            log.debug("Processed record value: {}", processedRecord.value());
+            // Send the processed record to the record sink
+            recordSink.emit(new SourceRecordAndResult(record, List.of(processedRecord), null));
+        } catch (Throwable e) {
+            log.error("Error processing record: {}", e.getMessage());
+            recordSink.emit(new SourceRecordAndResult(record, null, e));
+        }
+    }
+
+    @Override
+    protected Map<String, Object> buildAdditionalInfo() {
+        return Map.of("bucketName", bucketName);
+    }
+
+    private static class S3SourceRecord implements Record {
```

Comment:
we could:
1) rename this to S3ProcessorRecord
2) use the same class of the S3Source
3) switch to SimpleRecord